geo_map/static/geo_map/index.js

```diff
@@ -60,14 +61,86 @@ document.addEventListener("DOMContentLoaded", function () {
         if (providerSelect.loadOptions) {
           providerSelect.loadOptions();
         }
-        selectedTenants = providers.map((provider) => provider.value);
+        selectedTenants = providers.map((provider) => provider.id);
+        allVendors = JSON.parse(JSON.stringify(providers));
         initMap();
       })
       .catch((error) => console.error("Error fetching providers:", error));
   }
   fetchProviders();
 });
 
+async function fetchAndCreateMapData(
+  selectedPopsStatuses,
+  selectedGroups,
+  selectedTenants,
+  selectedFiberLinkStatuses
+) {
+  const currentZoom = map ? map.getZoom() : 6;
+  const currentCenter = map ? map.getCenter() : null;
+  storedMapCenter = currentCenter;
+  storedZoomLevel = currentZoom;
+
+  const SITES_API_CALL = new URL(baseURL + "/api/plugins/geo_map/sites/");
+  const LINKS_API_CALL = new URL(baseURL + "/api/plugins/geo_map/links/");
+
+  if (selectedPopsStatuses.length) {
+    SITES_API_CALL.searchParams.set("status__in", selectedPopsStatuses.join(","));
+  }
+  if (selectedGroups.length && selectedPopsStatuses.length) {
+    SITES_API_CALL.searchParams.set("?group__in", selectedGroups.join(","));
```

Comment:
typo? remove `?`

Reply:
I can't remove group__in because the API requires it to properly filter by groups, and removing it causes an error due to invalid query parameters